docs: document security chain change in 3.10.19 upgrade note (#836)
gravitee-io/gravitee-docs: pages/apim/3.x/installation-guide/upgrades/3.10.19/README.adoc

= Upgrade to 3.10.19

== Breaking changes

=== API key and JWT plans

Security chain parses all active plans to select and execute the one matching the incoming request.

**Before this version :**

- API key plan is executed if the request contains an API key
- JWT plan is executed if the request contains a Bearer token

**From this version :**

Security chain has been improved to select and execute plans more efficiently.

> That causes some behavior changes.

Reviewer: ```suggestion
```

> Api keys and JWT plans are executed only if there is an active subscription related to the provided security token.

Reviewer: ```suggestion
```

> If there is no relevant subscription, the plans won't be executed and security chain will continue the parse following plans.

Reviewer: ```suggestion
```

> Here are some concrete examples of behavior changes :

Reviewer: ```suggestion
Specific examples of this change are provided in the table below:
```

|===

> |API plans | Request |Before this version |From this version

Reviewer: ```suggestion
| API plans | Request | Prior to this version | From this version
```

|API key plan + Keyless plan
|Request contains an invalid API key

> |Api key plan is executed = HTTP 401 unauthorized

Reviewer: ```suggestion
|API key plan is executed = HTTP 401 unauthorized
```
